src/rosdistro/manifest_provider/__init__.py

```diff
@@ -33,13 +33,5 @@
 
 
 def get_release_tag(repo, pkg_name):
-    data = {
-        'package': pkg_name
-    }
-    if repo.version is not None:
-        data['version'] = repo.version
-        data['upstream_version'] = repo.version.split('-')[0]
-    release_tag = repo.tags['release']
-    for k, v in data.items():
-        release_tag = release_tag.replace('{%s}' % k, v)
-    return release_tag
+    """ Compatibility shim. """
```

Comment:
The docblock is not really helpful. I would suggest remove it again and adding a plain comment above the function mentioning that it is being used by `rosinstall_generator`.